app/core/process.py

```diff
@@ -726,3 +735,36 @@ def start_background_process(config_filepath):
         cam.teardown()  # Teardown the camera and stop it
         update_status_file(cam)  # Update the status file with halted status
     os.close(CameraCoreModel.fifo_fd)  # Close the FIFO pipe
+
+
+def execute_macro_command(script_name, args):
+    """
+    Executes a macro script located in /var/www/html/macros/ with specified arguments.
+
+    Args:
+        script_name (str): The name of the macro script file (e.g., "somemacro.sh").
+        args (list): List of arguments to pass to the script.
+    """
+    # Define the base directory for macros and construct the full script path
+    macros_dir = "/var/www/html/macros"
+    script_path = os.path.join(macros_dir, script_name)
+
+    # Check if the script exists and is executable
+    if not os.path.isfile(script_path):
+        print(f"ERROR: Script {script_path} does not exist.")
+        return False
+    if not os.access(script_path, os.X_OK):
+        print(f"ERROR: Script {script_path} is not executable.")
+        return False
+
+    # Construct the command with 'www-data' user context
+    command = ["sudo", "-u", "www-data", script_path] + args
```

Comment:
Sorry, I'm a bit tired. Would you mind rewriting that second point? Would you like me to not explicitly state the user context? If this is the case this line would be `command = [script_path] + args`.

Reply:
Also, no clue if I said this before. I'm coding blind here as I don't have access to a Pi for a bit. Sorry xD

Reply:
Yes, that's right. The code does not need to specify a user context.
(My feature request was a little misleading there...) :-)

I'll try to test your code when I get a chance!

Thanks again!

Reply:
Ah, no worries. Thanks for clarifying! 

I’ll add that change now. 

Take your time with testing 🙌